test/controllers/NirmsNumberControllerSpec.scala

```diff
@@ -0,0 +1,119 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import controllers.actions.FakeAuthoriseAction
+import forms.NirmsNumberFormProvider
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+import play.api.test.Helpers._
+import play.api.test.FakeRequest
+import views.html.NirmsNumberView
+
+class NirmsNumberControllerSpec extends SpecBase {
+
+  private val formProvider          = new NirmsNumberFormProvider()
+  private val nirmsNumberView       = app.injector.instanceOf[NirmsNumberView]
+  private val nirmsNumberController = new NirmsNumberController(
+    stubMessagesControllerComponents(),
+    new FakeAuthoriseAction(defaultBodyParser),
+    nirmsNumberView,
+    formProvider
+  )
+
+  "Nirms Number Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val result = nirmsNumberController.onPageLoad(fakeRequest)
+
+      status(result) mustEqual OK
+
+      contentAsString(result) mustEqual nirmsNumberView(formProvider())(fakeRequest, stubMessages()).toString
+
+    }
+
+    "must redirect on Submit when user enters correct Nirms number (GB region)" in {
+
+      val fakeRequestWithData = FakeRequest().withFormUrlEncodedBody("nirmsNumber" -> "RMS-GB-123456")
+
+      val result = nirmsNumberController.onSubmit(fakeRequestWithData)
+
+      status(result) mustEqual SEE_OTHER
+      //TO DO : Redirect needs to be changed to actual page when available
+      redirectLocation(result) shouldBe Some(routes.DummyController.onPageLoad.url)
+
+    }
+
+    "must redirect on Submit when user enters correct Nirms number (NI region)" in {
+
+      val fakeRequestWithData = FakeRequest().withFormUrlEncodedBody("nirmsNumber" -> "RMS-NI-123456")
+
+      val result = nirmsNumberController.onSubmit(fakeRequestWithData)
+
+      status(result) mustEqual SEE_OTHER
+      //TO DO : Redirect needs to be changed to actual page when available
+      redirectLocation(result) shouldBe Some(routes.DummyController.onPageLoad.url)
+
+    }
+
+    "must send bad request when user enters incorrect Nirms number" in {
+
+      val inCorrectnirmsNumber = "DDD-GB-123456"
+
+      val formWithErrors = formProvider().bind(Map("nirmsNumber" -> inCorrectnirmsNumber))
+
+      val fakeRequestWithData = FakeRequest().withFormUrlEncodedBody("nirmsNumber" -> inCorrectnirmsNumber)
+
+      val result = nirmsNumberController.onSubmit(fakeRequestWithData)
+
+      status(result) mustEqual BAD_REQUEST
+
+      contentAsString(result) mustEqual nirmsNumberView(formWithErrors)(fakeRequest, stubMessages()).toString
```

Comment:
I think we decided as a group to test the specific error scenarios in the unit tests for form failures, can we check that the error message returned for a form error is present in these e.g.
contentAsString(result) mustContain "the error message we return"

Reply:
Added the check for the message.